### PR TITLE
implement APIs, in 'db-as-argument' style and depend on coreword

### PR DESCRIPTION
**Pull request by complected**
~~Status: PR WIP, please wait, will update~~
Status: ready for review

- implement modular APIs: `make`, `create`, `get`, `ask`, `add`
- all APIs, except `make`, takes `db` as arg, so db can simply embedded test client or actual client
- take `coreword` as submodule, what are your thoughts on it?
- [ ] TODO: need to update submodule version, if and after that PR is merged - https://github.com/nmushegian/coreword/pull/5

Thanks in advance!

### CODE REVIEW

**Comment by complected** [comment]
```
TAP version 13
# make and use
ok 1 make - should make
ok 2 create - should create
ok 3 add - should add
ok 4 get - should be equal
# invalid sig
ok 5 make - should make
ok 6 create - should create
ok 7 show throw
# ask - prop: nom
ok 8 ask - should return 1 record only
ok 9 ask - should be equal
# ask - prop: wat
ok 10 ask - should return 1 record only
ok 11 ask - should be equal
# ask - prop: dat
ok 12 ask - should return 1 record only
ok 13 ask - should be equal
# ask - prop: exp
ok 14 ask - should return 1 record only
ok 15 ask - should be equal
# ask - prop: sig
ok 16 ask - should return 1 record only
ok 17 ask - should be equal
# ask - prop: ecr
ok 18 ask - should return 1 record only
ok 19 ask - should be equal

1..19
# tests 19
# pass  19

# ok

```

File: .gitmodules [hunk 1]
```diff
@@ -0,0 +1,3 @@
+[submodule "coreword"]
```
Comment (complected) [inline .gitmodules:1]: Thoughts on using gitmodule in this context? I was just thinking to skip NPM for coercion risk.

Comment (nmushegian) [reply .gitmodules:1]: I love submodules haha

File: test.js [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+const sk = ubye(sk_txt);
+const pk = pkey(sk);
+
+test('make and use', async t => {
+  const db = make();
+  t.ok(db, "make - should make");
+  t.ok(create(db), "create - should create");
+  const nom = `vitalik-${Math.floor(Math.random() * 10000)}`;
+  const wat = "https://ethresear.ch";
+  const dat = ":free:vitalik:dai";
```
Comment (nmushegian) [inline test.js:26]: Is this a real locked path haha

**Comment by nmushegian** [comment]
If this depends on your coreword changes then I have to close this, please make smaller PRs until we get on the same page

**Pull request closed without merging**